src/tmt300rs/hid-tmt300rs.c

```diff
@@ -948,6 +980,8 @@ static int t300rs_upload_constant(struct t300rs_device_entry *t300rs,
 	int ret;
 
 	level = (constant.level * fixp_sin16(effect.direction * 360 / 0x10000)) / 0x7fff;
+	/* the Windows driver uses the range [-16385;16381] */
```

Comment:
The previous comment also applies to this, even more weird is the range being asymmetric.

Reply:
There is an about `-2` offset on the whole range in the constant force magnitude and in the periodic effect offset.
The Windows driver using `-2` instead of `0` force/offset in these cases and never uses `0` (see spreadsheet).

I have not replicated this because I _think_ it's a rounding error and saw no difference in behavior.